lib/Frontend/CompilerInvocation.cpp

```diff
@@ -66,6 +66,11 @@ void CompilerInvocation::setMainExecutablePath(StringRef Path) {
       Path, FrontendOpts.UseSharedResourceFolder, LibPath);
   setRuntimeResourcePath(LibPath.str());
 
+  llvm::SmallString<128> clangPath(Path);
+  llvm::sys::path::remove_filename(clangPath);
+  llvm::sys::path::append(clangPath, "clang");
+  ClangImporterOpts.clangPath = std::string(clangPath);
```

Comment:
Do we need to worry about the file suffix?  This should be `clang.exe` if running on Windows

Reply:
We're only worried about the directory that the compiler is located in – Clang throws away the last component of the path. It happens here: https://github.com/apple/llvm-project/blob/c5c5f90552428d9012cc0339306eed06ac864e2a/clang/lib/Driver/Driver.cpp#L150]

Before this change, Swift always used `clang` without the `.exe` suffix, and it worked well for Windows.